### PR TITLE
add resource_type delegate

### LINKED ISSUE #1498: Users should be allowed to assert a resource type on collection
Missing Resource Type error on create a new Collection and view a collection

### PR DESCRIPTION
**Pull request by njaffer**
closes #1498

### CODE REVIEW

**Comment by jcoyne** [comment]
Please add a test to this PR.

**Comment by njaffer** [comment]
@jcoyne, the test already has :resource_type in the terms. I checked the work presenter specs and don't see any additional coverage for testing the delegate. Can you please specify what do you mean by adding a test?

**Comment by jcoyne** [comment]
@njaffer the test should check that the code you added works as you intend.  So I'd probably write a test like this:

```ruby
let(:presenter) { CollectionPresenter.new(solr_doc) }
let(:solr_doc) { double("Mock Solr document", resource_type: 'Collection') }
subject { presenter.resource_type }
it { is_expected.to eq 'Collection' }
```

You should have a test that fails before you make changes to the code and then succeeds after making the changes.  In this way no-one will accidentally revert your changes without seeing a test fail.

**Comment by njaffer** [comment]
@jcoyne, this code gives a lot of errors, can we just check the resource type as part of the terms? I have added the test for that, let me know what you think

@mjgiarlo, I looked at adding the other delegates, but did not seem to justify adding all the extra ones when they are not needed. May be I am missing something?

File: spec/presenters/sufia/collection_presenter_spec.rb [hunk 1]
```diff
@@ -8,14 +8,19 @@
                             :language, :identifier, :based_near, :related_url] }
   end
 
-  let(:collection) { Collection.new(id: '111', description: 'a nice collection', title: 'A clever title') }
-  let(:solr_doc) { SolrDocument.new(collection.to_solr) }
+  let(:collection) { Collection.new(id: '111', description: 'a nice collection', title: 'A clever title', resource_type: ['Collection']) }
   let(:ability) { double }
   let(:presenter) { described_class.new(solr_doc, ability) }
+  let(:solr_doc) { SolrDocument.new(collection.to_solr, resource_type: 'Collection') }
```
Comment (jcoyne) [inline spec/presenters/sufia/collection_presenter_spec.rb]: is it necessary to provide `resource_type: 'Collection'` here?  Shouldn't `collection.to_solr` take care of that?

**Comment by mjgiarlo** [comment]
@njaffer: I agree that we shouldn't add the others until/unless we're sure they're necessary.

**Comment by njaffer** [comment]
@jcoyne , yep don't need resource_type: 'Collection'. Updated the PR

**Pull request merged**